lib/charms/layer/hacluster.py

```diff
@@ -0,0 +1,28 @@
+from charms.reactive import clear_flag, set_flag
+
+services = {}
+
+
+def add_service_to_hacluster(name, service_name):
+    """Adds a service to be monitored under HAcluster.
+       Takes a name for this entry and a service_name,
+       which is the name used by systemd/initd to identify
+       the service.
+    """
+    global services
+
+    services[name] = service_name
```

Comment:
`services` is not persisted between hook invocations. You probably want to use `charmhelpers.core.unitdata.kv` to persist the data. [Docs](https://charm-helpers.readthedocs.io/en/latest/api/charmhelpers.core.unitdata.html)

Reply:
done, thanks. I came to this conclusion at some point and thought I had fixed that, but obviously not.